src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -207,6 +207,74 @@ public static Array CreateInstance(Type elementType, params long[] lengths)
             return CreateInstance(elementType, intLengths);
         }
 
+        public static unsafe Array CreateInstanceFromArrayType(Type arrayType, int length)
+        {
+            ArgumentNullException.ThrowIfNull(arrayType);
+            ArgumentOutOfRangeException.ThrowIfNegative(length);
+
+            if (!arrayType.IsArray)
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Arg_MustBeArrayType, ExceptionArgument.arrayType);
+
+            RuntimeType t = (arrayType.GetElementType()!.UnderlyingSystemType as RuntimeType)!;
+            Debug.Assert(t is not null, $"CreateInstanceFromArrayType can not get underlying system type for \"{arrayType}\"");
```

Comment:
The `t` variable can be defined as nullable to save a `!`. I think we are allowing `var` here.

Reply:
`t` can not be null here so I use `RuntimeType`, not `RuntimeType?`.